src/main/java/common/tileentities/GTMTE_FluidMultiStorage.java

```diff
@@ -59,9 +60,26 @@ public IMetaTileEntity newMetaEntity(IGregTechTileEntity var1) {
 		return new GTMTE_FluidMultiStorage(super.mName);
 	}
 
-	@Override
-	public String[] getDescription() {
-		final GT_Multiblock_Tooltip_Builder tt = new GT_Multiblock_Tooltip_Builder();
+    @Override
+    public String[] getDescription() {
+        return getCurrentDescription();
+    }
+
+    @Override
+    public boolean isDisplaySecondaryDescription() {
+        return Keyboard.isKeyDown(Keyboard.KEY_LSHIFT);
+    }
+
+    public String[] getPrimaryDescription() {
+        return getTooltip().getInformation();
+    }
+
+    public String[] getSecondaryDescription() {
+        return getTooltip().getStructureInformation();
+    }
+
+    protected GT_Multiblock_Tooltip_Builder getTooltip() {
+        final GT_Multiblock_Tooltip_Builder tt = new GT_Multiblock_Tooltip_Builder();
```

Comment:
I'd prefer to make this cached in a static final field.

Reply:
My initial intention is to modify the code as little as possible, the original code did not cache tooltip
should have been changed